codemon/CodemonMeta.py

```diff
@@ -34,3 +38,52 @@ def get_filename():
 def get_practice_files():
   practiceFiles = ['A.cpp','B.cpp','C.cpp']
   return practiceFiles
+
+def write_to_file(filename, text, contestName=None):
+  full_filename = os.path.join(os.getcwd(), filename)
+  if contestName is not None:
+    full_filename = os.path.join(os.getcwd(), contestName, filename)
+  with open(full_filename, 'w+') as f:
+    f.write(text)
+
+def compile_and_run(filename):
+  # Store full file paths
+  full_filename = os.path.join(os.getcwd(), filename)
+  full_output_filename = os.path.join(os.getcwd(), 'prog')
+  full_input_filename = os.path.join(os.getcwd(), 'input.txt')
+
+  # Check if required files exist
+  if not Path(full_filename).is_file():
+    print(f"{filename} doesn't exist.", file=sys.stderr)
+  if not Path(full_input_filename).is_file():
+    print(f"'input.txt' doesn't exist.", file=sys.stderr)
+
+  compilation_child_process = subprocess.Popen(['g++', full_filename, '-o', full_output_filename], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
+  compilation_child_process.wait()
+  compilation_child_process.terminate()
+  if compilation_child_process.returncode != 0:
+    return_code = compilation_child_process.returncode
+    if return_code == 127:
+      print("'g++' isn't installed.", file=sys.stderr)
+    else:
+      print('Error in compilation.', file=sys.stderr)
+      print(f'Return code: {return_code}', file=sys.stderr)

```

Comment:
We can also omit these lines here, since the compilation error already gets displayed.